pkg/cmd/roachtest/roachtestutil/mixedversion/steps.go

```diff
@@ -715,3 +715,51 @@ func startStopOpts(opts ...option.StartStopOption) []option.StartStopOption {
 		option.NoBackupSchedule,
 	}, opts...)
 }
+
+type panicNodeStep struct {
+	rt         test.Test
+	initTarget int
+}
+
+func (s panicNodeStep) Background() shouldStop { return nil }
+
+func (s panicNodeStep) Description() string { return "run panic node" }
+
+func (s panicNodeStep) Run(ctx context.Context, l *logger.Logger, rng *rand.Rand, h *Helper) error {
+	nodeList := h.System.Descriptor.Nodes
+	targetNode := nodeList[rng.Intn(len(nodeList))]
+	nodeVersion, err := h.System.NodeVersion(targetNode)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to get node version from target node %d", targetNode)
+	}
+
+	binaryPath := clusterupgrade.CockroachPathForVersion(s.rt, nodeVersion)
+	settings := install.MakeClusterSettings(
+		install.BinaryOption(binaryPath),
+		install.TagOption(systemTag),
+	)
+	customStartOpts := []option.StartStopOption{option.WithInitTarget(s.initTarget)}
```

Comment:
We wanna do `option.SkipInit` here. There might be more nuances so double check with `restartWithNewBinaryStep` to see if there's anything else.

Consider splitting the panic logic and the restarting logic into two different steps. I think this gives a few benefits:

1. We will have a standardized "restart" step to maintain. As you add more failure mutators, you can restart nodes using the same step instead of having to reimplement it and potentially missing things like this.
2. It sets you up to allow for failures to be recovered some arbitrary n steps after injection instead of immediately after.
3. nit: it makes for a clearer description that we are not just panicking a node, but restarting it